### DISCUSSION TITLE
Incremental Gradle build

### DISCUSSION THREAD

**Discussion by filipblondeel** [ImFlog/schema-registry-plugin#178] [Q&A] [answered]
Hello,

I'm currently using this plugin to download Avro schemas from a Confluent Schema Registry. To ensure my Gradle build runs as fast as possible, I would like to avoid running the `downloadSchemasTask` task if the inputs and outputs don't change. The problem I'm facing is that I don't know how to do that. Out of the box, the task is never 'up-to-date' and is always executed.

Could you help me?

**Comment by ImFlog** [comment, marked as answer]
Hello,

Normally we are using tasks with Inputs so if the inputs don't change it shouldn't replay the tasks.
I probably miss configured something and need to investigate a bit.
Could you create a reproduction repository and submit an issue to let me investigate ?

Thank you

**Reply by filipblondeel** [reply]
I created #179 and linked a sample repository with a minimal Gradle build to reproduce the issue.
